Test/dafny0/ConcurrentAttribute.dfy

```diff
@@ -0,0 +1,56 @@
+// RUN: %exits-with 4 %verify --reads-clauses-on-methods "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+class Box<T> {
+  var x: T
+
+  constructor(x: T)
+    reads {}
+  {
+    this.x := x;
+  }
+}
+
+class Concurrent {
+
+  function {:concurrent} GoodFn(b: Box<int>): int {
+    42
+  }
+
+  function {:concurrent} BadFn(b: Box<int>): int  // Error: reads clause might not be empty ({:concurrent} restriction)
+    reads b
+  {
+    b.x
+  }
+
+  function {:concurrent} WeirdButOkayFn(b: Box<int>): int 
+    reads if false then {b} else {}
+  {
+    42
+  }
+
+  method {:concurrent} GoodM(b: Box<int>) {
+  }
+
+  method {:concurrent} BadM(b: Box<int>)  // Error: reads clause might not be empty ({:concurrent} restriction)
+    reads b
+  {
+    var x := b.x;
+  }
+
+  method {:concurrent} WeirdButOkayM(b: Box<int>) 
```

Comment:
Please add a test where the reads clause has an empty object set but a designed field like reads {}`f (something that compiles of course)
Also, at least for future change detection, please add a test with a reads a `f and another with modifies a`f where f is a ghost field. You decided that it's not concurrent safe for this PR, but since I exhibited a sound concurrent execution model where ghost fields can be read and modified independently by threads, we should make sure we differentiate against that.